### ISSUE TITLE
some undefined reference errors in demo apps

### ISSUE THREAD

**Issue by qingniaotonghua** [part-cw/lambdanative#53]
When I make the demo app named 'DemoCamera', the error accured :
https://gist.github.com/qingniaotonghua/1792fec7663743e4126a

When I make the demo app named 'packtool', the error accured (the demo pngtool, ttftool, DemoConsole, DemoFastCGI, DemoSchemeAndC also appeared this kind error):
https://gist.github.com/qingniaotonghua/143e0b9ce567b952cbd4

When I make the demo app named 'DemoSensors', the error accured :
https://gist.github.com/qingniaotonghua/a8347887ac4198d27893

When I make the demo app named 'DemoJarFiles', the error accured :
https://gist.github.com/qingniaotonghua/b42a32894715a50f4365

When I make the demo app named 'DemoSerial', the error accured :
https://gist.github.com/qingniaotonghua/a0d6b7b2318253128660

**Comment by mgorges** [comment]
There are different problems in your issue: 
a) packtool, pngtool, ttftool, DemoConsole, DemoFastCGI, and DemoSchemeAndC are STANDALONE apps, which won't compile on android. This is as the C-hook doesn't contain the ffi_event function as there is no main application loop. However, the Android system depends on this hook to drive its render loop so compilation fails.
b) The other errors are puzzling. as it seems to be that your android build is complete. Could you try running this one as `./configure DemoSensors android verbose; make` and paste the results as a Gist?

**Comment by qingniaotonghua** [comment]
Thanks a lot for your answer，the result is below:
When I make the demo app named 'DemoCamera' using "./configure DemoSerial android verbose;make"
the error accured :
https://gist.github.com/qingniaotonghua/42b97ee48a1c27828920

When I make the demo app named 'DemoSensors' using "./configure DemoSensors android verbose;make", the error accured :
https://gist.github.com/qingniaotonghua/e67bd1330a91452b454c

When I make the demo app named 'DemoJarFiles' using "./configure DemoJarFiles android verbose;make", the error accured :
https://gist.github.com/qingniaotonghua/d06087ed7548ccd9c78f

When I make the demo app named 'DemoSerial' using "./configure DemoSerial android verbose;make", the error accured :
https://gist.github.com/qingniaotonghua/d3dfd04fe2718149797e

**Comment by mgorges** [comment]
I think we are having a Unicode problem somwhere in our build infrastructure! If I copy your username I can't build these applicatiions either. This may require some digging to solve but I will let you know when we fixed this. Sorry.

**Comment by mgorges** [comment]
Please try this patch and report back:
```
diff --git a/tools/subtool/subtool.c b/tools/subtool/subtool.c
index 124a6b5..2a4cc4f 100755
--- a/tools/subtool/subtool.c
+++ b/tools/subtool/subtool.c
@@ -71,7 +71,7 @@ void load_pattern(char *fname)
       fgets(buf,1024,fd);
       if (feof(fd)||ferror(fd)||buf[0]==0) break;
       l=strlen(buf);
-      for (i=0;i<l;i++) { if (buf[i]<32) buf[i]=0; }
+      for (i=0;i<l;i++) { if (buf[i]>0 && buf[i]<32) buf[i]=0; }
       for (i=1;i<l;i++) { if (buf[i]==buf[0]) break; }
       buf[i]=0;
       struct pattern *tmp = (struct pattern*)malloc(sizeof(struct pattern));
```

**Comment by mgorges** [comment]
Fixed in 76595e4? - If not please reopen, thanks.

**Comment by qingniaotonghua** [comment]
Thanks sooooo much for your excellent help, the demo apps can run normally...
